src/js/background.js

```diff
@@ -87,6 +87,12 @@ listen((port, msg) => {
       port.postMessage(['NOT JSON', 'technically JSON but not an object or array']);
       port.disconnect();
       return;
+    } 
+    // If there's an empty object or array, return JSON as is
+    else if (Object.entries(obj).length === 0 || obj.length === 0) {
```

Comment:
Wouldn't `obj.length === 0` be sufficient?

Reply:
Thank you for the suggestion. `obj.length === 0` would only work for the array since arrays have a `length` property, but not [objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object). (This can also be verified by trying `object.hasOwnProperty('length');`

Since `[Object.entries()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries)` returns an array of key value pairs, we can use that to see if an object has any content. (you could do this with `.keys()` too. I tend to use `.entries()` for semantics).